### PR TITLE
move from Glide to dep

### PR DESCRIPTION
**Pull request by bassam**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,5 @@
 
 /_output
 /vendor
+/_vendor*/
+/vendor.orig/
```
Comment (kokhang) [inline .gitignore]: add a new line here

Comment (paha) [inline .gitignore]: Do we need these two entries? Looks like they are a user use case.

Do we need to add any of dep other files like - https://github.com/golang/dep/blob/master/.gitignore

File: build/makelib/golang.mk [hunk 2]
```diff
@@ -161,23 +156,15 @@ go.fmt:
...
-	@echo === installing glide
-	@mkdir -p $(TOOLS_HOST_DIR)/tmp
-	@curl -sL https://github.com/Masterminds/glide/releases/download/$(GLIDE_VERSION)/glide-$(GLIDE_VERSION)-$(GOHOSTOS)-$(GOHOSTARCH).tar.gz | tar -xz -C $(TOOLS_HOST_DIR)/tmp
-	@mv $(TOOLS_HOST_DIR)/tmp/$(GOHOSTOS)-$(GOHOSTARCH)/glide $(GLIDE)
-	@rm -fr $(TOOLS_HOST_DIR)/tmp
+$(DEP):
+	@echo === installing dep
+	@mkdir -p $(TOOLS_HOST_DIR)
+	@curl -sL -o $(DEP) https://github.com/golang/dep/releases/download/$(DEP_VERSION)/dep-$(GOHOSTOS)-$(GOHOSTARCH)
+	@chmod +x $(DEP)
```
Comment (paha) [inline build/makelib/golang.mk]: Generally, it is not a good idea to install publicly available software on user workstations without asking. Should the make validate if the tool installed and if not error out with a message for instructions on what is required?

Comment (bassam) [reply build/makelib/golang.mk]: It’s installed in a private path. Our approach has been to install tools needed for the build in a private path that does not interfere with the host

File: .gitignore (hunk 1 above)
Comment (bassam) [reply .gitignore]: Ok

Comment (bassam) [reply .gitignore]: Vendor.orig seems to run every time rep ensure runs. The other one seems to be used during migration but it’s harmless to leave here. I could take it out

Comment (paha) [reply .gitignore]: You are right, they are harmless, didn't realize `dep` as creating those.

File: build/makelib/golang.mk (hunk 2 above)
Comment (paha) [reply build/makelib/golang.mk]: got it, thanks for the explanation.

**Pull request merged**